src/net_processing.cpp

```diff
@@ -2442,18 +2441,18 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         CAddress addrMe;
         CAddress addrFrom;
         uint64_t nNonce = 1;
-        uint64_t nServiceInt;
         ServiceFlags nServices;
         int nVersion;
         std::string cleanSubVer;
         int starting_height = -1;
         bool fRelay = true;
 
-        vRecv >> nVersion >> nServiceInt >> nTime >> addrMe;
+        vRecv >> nVersion >> Using<CustomUintFormatter<8>>(nServices) >> nTime;
         if (nTime < 0) {
             nTime = 0;
         }
-        nServices = ServiceFlags(nServiceInt);
+        // The pre-31402 serialization of addrMe (which didn't include nTime):
+        vRecv >> Using<CustomUintFormatter<8>>(addrMe.nServices) >> static_cast<CService&>(addrMe);
```

Comment:
The service bits here are never used (we don't care what the peer thinks our services are), so we can be more explicit in discarding them and only using the `CService` data:

```diff
--- a/src/net_processing.cpp
+++ b/src/net_processing.cpp
@@ -2438,7 +2438,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         }
 
         int64_t nTime;
-        CAddress addrMe;
+        CService addrMe;
         CAddress addrFrom;
         uint64_t nNonce = 1;
         ServiceFlags nServices;
@@ -2451,8 +2451,10 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         if (nTime < 0) {
             nTime = 0;
         }
-        // The pre-31402 serialization of addrMe (which didn't include nTime):
-        vRecv >> Using<CustomUintFormatter<8>>(addrMe.nServices) >> static_cast<CService&>(addrMe);
+        // Ignore the addrMe service bits sent by the peer
+        uint8_t unused_service_bits[8];
+        vRecv >> unused_service_bits;
+        vRecv >> addrMe;
         if (!pfrom.IsInboundConn())
         {
             m_addrman.SetServices(pfrom.addr, nServices);
```

Reply:
Done (using `vRecv.ignore(8);` instead).